pkg/nsx/services/inventory/builder.go

```diff
@@ -277,17 +278,24 @@ func (s *InventoryService) BuildNamespace(namespace *corev1.Namespace) (retry bo
 		preContainerProject = *preContainerProject.(*containerinventory.ContainerProject)
 	}
 
-	// Extract network errors from annotations
+	// Extract network errors from namespace conditions
 	// Initialize as empty slice to ensure NSX receives [] instead of null when clearing errors
 	networkErrors := make([]common.NetworkError, 0)
 	networkStatus := NetworkStatusHealthy
-
-	// Check for VPC error annotation
-	if errorMsg, exists := namespace.Annotations[ctrcommon.AnnotationNamespaceVPCError]; exists && errorMsg != "" {
-		networkErrors = append(networkErrors, common.NetworkError{
-			ErrorMessage: errorMsg,
-		})
-		networkStatus = NetworkStatusUnhealthy
+	for _, condition := range namespace.Status.Conditions {
+		if condition.Type == networkinfo.NamespaceNetworkReady && condition.Status == corev1.ConditionFalse {
+			// Create a network error with the reason and message
+			errorMessage := condition.Reason + ": " + condition.Message
+			networkError := common.NetworkError{
+				ErrorMessage: errorMessage,
+			}
+			networkErrors = append(networkErrors, networkError)
+			networkStatus = NetworkStatusUnhealthy
+		} else if condition.Type == networkinfo.NamespaceNetworkReady && condition.Status == corev1.ConditionTrue {
+			networkErrors = []common.NetworkError{}
```

Comment:
should we use make([]common.NetworkError,0) here?

Reply:
checked with doc
- - More explicit about creating an empty slice with specific length `make([]common.NetworkError, 0)`
- - More concise and idiomatic Go `[]common.NetworkError{}`
no difference